os/tools/compression/src/compress.c

```diff
@@ -65,6 +66,7 @@ static void compress_file(int block_size, int type, char *in_file, char *out_fil
 #if CONFIG_COMPRESSION_TYPE == LZMA
 	long unsigned int propsSize = LZMA_PROPS_SIZE;
 #endif
+	bool compression_fail = false;
```

Comment:
It's not related to this file but below 'stat' with same parameter seems that already checked in main function.

Reply:
As you have suggested, I have removed the stat for the input file as it is already checked in the main function before this function is called.

Reply:
All `goto error` cases are compression fail.
So I think we don't need to set and use this variable.
We just call `exit()` from goto statement.

Reply:
Hello @jeongchanKim, calling exit from goto statement causes successful compression cases also to fail as the statements in error are executed every time. Please let me know if I did understand your comment correctly. So I think it's better to use this boolean variable and call exit only during the fail cases.

Reply:
Hello @Taejun-Kwon , the stat buf here for the input file is used to get the size, hence I am not deleting the above lines. I will revert the changes made.

Reply:
Because of naming, we feel they `goto error` and `compression_fail = true;` look duplicated. How about renaming `error:` to `exit:` or `finish:` or something like that?

Reply:
Done. I have changed error to exit.

Reply:
Okay. I understood. Thanks.